src/app/core/data/relationship.service.ts

```diff
@@ -254,20 +228,113 @@ export class RelationshipService extends DataService<Relationship> {
+    return this.getItemRelationshipsArray(item).pipe(
+      switchMap((relationships: Relationship[]) => {
+        return observableCombineLatest(...relationships.map((relationship: Relationship) => {
+          const isLeftItem$ = this.isItemInUUIDArray(relationship.leftItem, uuids);
+          const isRightItem$ = this.isItemInUUIDArray(relationship.rightItem, uuids);
+          return observableCombineLatest(isLeftItem$, isRightItem$).pipe(
+            filter(([isLeftItem, isRightItem]) => isLeftItem || isRightItem),
+            map(() => relationship),
+            startWith(undefined)
+          );
+        }))
+      }),
+      map((relationships: Relationship[]) => relationships.filter(((relationship) => hasValue(relationship)))),
+    )
+  }
+
+  private isItemInUUIDArray(itemRD$: Observable<RemoteData<Item>>, uuids: string[]) {
+    return itemRD$.pipe(
       getSucceededRemoteData(),
-      flatMap((rd: RemoteData<Relationship>) => observableCombineLatest(rd.payload.leftItem.pipe(getSucceededRemoteData()), rd.payload.rightItem.pipe(getSucceededRemoteData()))),
-      take(1)
-    ).subscribe(([leftItem, rightItem]) => {
-      this.objectCache.remove(leftItem.payload.self);
-      this.objectCache.remove(rightItem.payload.self);
-      this.requestService.removeByHrefSubstring(leftItem.payload.self);
-      this.requestService.removeByHrefSubstring(rightItem.payload.self);
-    });
+      map((itemRD: RemoteData<Item>) => itemRD.payload),
+      map((item: Item) => uuids.includes(item.uuid))
+    );
+  }
+
+  getRelationshipByItemsAndLabel(item1: Item, item2: Item, label: string): Observable<Relationship> {
+    return this.getItemRelationshipsByLabel(item1, label)
+      .pipe(
+        getSucceededRemoteData(),
+        isNotEmptyOperator(),
+        map((relationshipListRD: RemoteData<PaginatedList<Relationship>>) => relationshipListRD.payload.page),
+        mergeMap((relationships: Relationship[]) => {
+          return observableCombineLatest(...relationships.map((relationship: Relationship) => {
+            return observableCombineLatest(
+              this.isItemMatchWithItemRD(relationship.leftItem, item2),
+              this.isItemMatchWithItemRD(relationship.rightItem, item2)
+            ).pipe(
+              map(([isLeftItem, isRightItem]) => isLeftItem || isRightItem),
+              map((isMatch) => isMatch ? relationship : undefined)
+            );
+          }))
+        }),
+        map((relationships: Relationship[]) => relationships.find(((relationship) => hasValue(relationship))))
+      )
+  }
+
+  private isItemMatchWithItemRD(itemRD$: Observable<RemoteData<Item>>, itemCheck: Item): Observable<boolean> {
+    return itemRD$.pipe(
+      getSucceededRemoteData(),
+      map((itemRD: RemoteData<Item>) => itemRD.payload),
+      map((item: Item) => item.uuid === itemCheck.uuid)
+    );
+  }
+
+  public setNameVariant(listID: string, itemID: string, nameVariant: string) {
```

Comment:
Please add Typedocs or a basic comment describing each of these public methods (there are 5 more below this one).   Most of these are pretty straightforward by name, but it's immediately clear what `listID` refers to in this method.